src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+    py               = std::min(static_cast<uint16_t>(100), py);
+    dspeed_threshold = std::min(static_cast<uint16_t>(100), dspeed_threshold);
+    // save values
+    state.senv_x_val       = px;
+    state.senv_y_val       = py;
+    state.dspeed_threshold = dspeed_threshold;
+    if ((px != 0) && (py != 0)) {
+        px--; // Inspired by CuteMouse
+        py--; // Although their cursor update routine is far more
+              // complex then ours
+        state.senv_x = (static_cast<float>(px) * px) / 3600.0f + 1.0f / 3.0f;
+        state.senv_y = (static_cast<float>(py) * py) / 3600.0f + 1.0f / 3.0f;
+    }
+}
+
+static void SetInterruptRate(uint16_t rate_id)
+{
+    switch (rate_id) {
+    case 0: rate_hz  = 0;   break; // no events, TODO: this should be simulated
+    case 1: rate_hz  = 30;  break;
+    case 2: rate_hz  = 50;  break;
+    case 3: rate_hz  = 100; break;
+    default: rate_hz = 200; break; // above 4 is not suported, set max
+    }
+
+    // Update event queue settings
+    if (rate_hz) {
+        // Update event queue settings
+        MOUSE_NotifyRateDOS(rate_hz);
+    }
+}
+
+static void ResetHardware()
+{
+    wheel = 0;
+    SetInterruptRate(4);
+    PIC_SetIRQMask(12, false); // lower IRQ line
+}
+
+void MOUSEDOS_BeforeNewVideoMode()
+{
+    if (CurMode->type != M_TEXT)
+        RestoreCursorBackground();
+    else
+        RestoreCursorBackgroundText();
+
+    state.hidden     = 1;
+    state.oldhidden  = 1;
+    state.background = false;
+}
+
+// TODO: Does way to much. Many things should be moved to mouse reset one day
+void MOUSEDOS_AfterNewVideoMode(const bool setmode)
+{
+    state.inhibit_draw = false;
+    // Get the correct resolution from the current video mode
+    uint8_t mode = mem_readb(BIOS_VIDEO_MODE);
+    if (setmode && mode == state.mode)
+        LOG(LOG_MOUSE, LOG_NORMAL)
+    ("New video mode is the same as the old");
+    state.gran_x = 0xffff;
+    state.gran_y = 0xffff;
+    switch (mode) {
+    case 0x00:
+    case 0x01:
+    case 0x02:
+    case 0x03:
+    case 0x07: {
+        state.gran_x = (mode < 2) ? 0xfff0 : 0xfff8;
+        state.gran_y = 0xfff8;
+        Bitu rows    = IS_EGAVGA_ARCH
+                             ? real_readb(BIOSMEM_SEG, BIOSMEM_NB_ROWS)
+                             : 24;
+        if ((rows == 0) || (rows > 250))
+            rows = 25 - 1;
+        state.maxpos_y = static_cast<int16_t>(8 * (rows + 1) - 1);
+        break;
+    }
+    case 0x04:
+    case 0x05:
+    case 0x06:
+    case 0x08:
+    case 0x09:
+    case 0x0a:
+    case 0x0d:
+    case 0x0e:
+    case 0x13:
+        if (mode == 0x0d || mode == 0x13)
+            state.gran_x = 0xfffe;
+        state.maxpos_y = 199;
+        break;
+    case 0x0f:
+    case 0x10: state.maxpos_y = 349; break;
+    case 0x11:
+    case 0x12: state.maxpos_y = 479; break;
+    default:
+        LOG(LOG_MOUSE, LOG_ERROR)
+        ("Unhandled videomode %X on reset", mode);
```

Comment:
wonky formatting

Reply:
Clang formatter, probably wanted to enforce maximum line length. Fixed.

Reply:
Yeah, sometimes it needs a bit of manual help...